src/test/java/se/sundsvall/billingdatacollector/api/CollectorResourceTest.java

```diff
@@ -0,0 +1,142 @@
+package se.sundsvall.billingdatacollector.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.springframework.http.MediaType.APPLICATION_PROBLEM_JSON;
+import static org.zalando.problem.Status.BAD_REQUEST;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.zalando.problem.ThrowableProblem;
+
+import se.sundsvall.billingdatacollector.Application;
+import se.sundsvall.billingdatacollector.service.CollectorService;
+
+@SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CollectorResourceTest {
+
+	@Autowired
+	private WebTestClient webTestClient;
+
+	@MockBean
+	private CollectorService mockService;
+
+	@Test
+	void testTriggerBilling() {
+		//Arrange
+		var flowInstanceId = "123";
+		doNothing().when(mockService).trigger(flowInstanceId);
+
+		//Act
+		webTestClient.post()
+			.uri(uriBuilder -> uriBuilder.path("/trigger/{flowInstanceId}")
+				.build(flowInstanceId))
+			.exchange()
+			.expectStatus().isAccepted();
+
+		//Assert
+		verify(mockService).trigger(flowInstanceId);
+		verifyNoMoreInteractions(mockService);
+	}
+
+	@Test
+	void testTriggerBillingBetweenTwoValidDates() {
+		//Arrange
+		doNothing().when(mockService).triggerBetweenDates(Mockito.any(LocalDate.class), Mockito.any(LocalDate.class));
+
+		var startDate = LocalDate.of(2023, 4, 25);
+		var endDate = LocalDate.of(2024, 4, 25);
+
+		//Act
+		webTestClient.post()
+			.uri(uriBuilder -> uriBuilder.path("/trigger")
+				.queryParam("startDate", startDate.toString())
```

Comment:
Maybe explicitly use the desired date format, to be absolutely sure that relying on default date formatting doesn't f*ck things up?

Reply:
Hm I think `localDate.toString()` will output on ISO-8601-format which is `yyyy-MM-dd` ?

Reply:
You are correct, dear sir

Reply:
I think I can remove the `toString` though.